File: src/nomanssave/j.java
package nomanssave;

import javax.swing.ComboBoxModel;
import javax.swing.event.ListDataListener;

class j implements ComboBoxModel {
   private eB A;
   // $FF: synthetic field
   final h z;

   j(h var1) {
      this.z = var1;
      this.A = null;
   }

   public int getSize() {
      return h.e(this.z).size();
   }

   public eB b(int var1) {
      return (eB)h.e(this.z).get(var1);
   }

   public void addListDataListener(ListDataListener var1) {
   }

   public void removeListDataListener(ListDataListener var1) {
   }

   public void setSelectedItem(Object var1) {
      this.A = (eB)var1;
      h.f(this.z);
   }

   public Object getSelectedItem() {
      return this.A;
   }

   // $FF: synthetic method
   public Object getElementAt(int var1) {
      return this.b(var1);
   }
}

File: NMSSaveEditor/nomanssave/upper/J.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Windows.Forms;
using System.Globalization;

namespace NMSSaveEditor
{

public class J : ComboBoxModel {
   public gh bs;
   // $FF: synthetic field
   public I bt;

   public J(I var1) {
      this.bt = var1;
      this.bs = null;
   }

   public int getSize() {
      return I.a(this.bt) == null ? 0 : I.a(this.bt).cD().Count;
   }

   public gh o(int var1) {
      return I.a(this.bt) == null ? null : (gh)I.a(this.bt).cD().Get(var1);
   }

   public void addListDataListener(ListDataListener var1) {
   }

   public void removeListDataListener(ListDataListener var1) {
   }

   public void setSelectedItem(Object var1) {
      this.bs = (gh)var1;
      if (this.bs == null) {
         I.b(this.bt).SetText("");
         I.c(this.bt).SetText("");
         I.c(this.bt).SetEnabled(false);
      } else {
         gy var2 = this.bs.cJ();
         I.b(this.bt).SetText(var2 == null ? "" : var2.ToString());
         I.c(this.bt).SetText(this.bs.cK());
         I.c(this.bt).SetEnabled(true);
      }

   }

   public Object getSelectedItem() {
      return this.bs;
   }

   // $FF: synthetic method
   public Object getElementAt(int var1) {
      return this.o(var1);
   }
}

}
